File: cpp/program/Produk.cpp
#include <iostream>
using namespace std;

/* Class Produk untuk menyimpan data produk di supermarket */
class Produk {
    protected:
        string kodeProduk;   // Kode unik produk
        string nama;         // Nama produk
        string harga;        // Harga produk (disimpan sebagai string)
        int stok;            // Jumlah stok produk

    public:
        // Constructor kosong
        Produk() {
            kodeProduk = "";
            nama = "";
            harga = "";
            stok = 0;
        }

        // Constructor lengkap
        Produk(string kodeProduk, string nama, string harga, int stok) {
            this->kodeProduk = kodeProduk;
            this->nama = nama;
            this->harga = harga;
            this->stok = stok;
        }

        // Getter dan Setter
        string getKodeProduk() {
            return kodeProduk;
        }

        void setKodeProduk(string kodeProduk) {
            this->kodeProduk = kodeProduk;
        }

        string getNama() {
            return nama;
        }

        void setNama(string nama) {
            this->nama = nama;
        }

        string getHarga() {
            return harga;
        }

        void setHarga(string harga) {
            this->harga = harga;
        }

        int getStok() {
            return stok;
        }

        void setStok(int stok) {
            this->stok = stok;
        }
        ~Produk(){}
};

File: python/program/Produk.py
# class Produk
class Produk:

    def __init__(self, kodeProduk: str, nama: str, harga: str, stok: int):
        self._kodeProduk = str(kodeProduk)
        self._nama = str(nama)
        self._harga = str(harga)   # disimpan string biar konsisten sama saldo/gaji
        self._stok = int(stok)

    # Getter & Setter Kode Produk
    def getKodeProduk(self) -> str:
        return self._kodeProduk

    def setKodeProduk(self, kodeProduk: str) -> None:
        self._kodeProduk = str(kodeProduk)

    # Getter & Setter Nama
    def getNama(self) -> str:
        return self._nama

    def setNama(self, nama: str) -> None:
        self._nama = str(nama)

    # Getter & Setter Harga
    def getHarga(self) -> str:
        return self._harga

    def setHarga(self, harga: str) -> None:
        self._harga = str(harga)

    # Getter & Setter Stok
    def getStok(self) -> int:
        return self._stok

    def setStok(self, stok: int) -> None:
        self._stok = int(stok)
